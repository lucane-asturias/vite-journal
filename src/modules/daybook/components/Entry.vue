<template>
  <div class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id }})"
  >
    <div class="entry-title d-flex">
      <span class="text-h6 font-weight-bold">{{ entryDay }}</span>
      <span class="mx-1 text-h6">{{ entryMonth }}</span>
      <span class="mx-2 text-font-weight-thin">{{ entryYear }}</span>
    </div>

    <div class="entry-description my-2">
      {{ shortText }}
    </div>
  </div>
</template>

<script setup>
  import { computed } from 'vue'

  const props = defineProps({
    entry: {
      type: Object,
      required: true
    }
  })

  const shortText = computed(() => {
    return props.entry.text.length > 135
      ? props.entry.text.substring(0, 135) + '...'
      : props.entry.text
  })

  const entryDay = computed(() => {
    return new Date(props.entry.date).getDate()
  }) 
   
  const entryMonth = computed(() => {
    return new Date(props.entry.date).toLocaleString("default", { month: "long" })
  })
      
  const entryYear = computed(() => {
    const options = { year: "numeric", weekday: "long" }
    const year_and_week = new Date(props.entry.date).toLocaleString("default", options)
    return year_and_week.substring(0, 4) + ',' + year_and_week.substring(4) // add a comma between 
  })

</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 42) !important;
    transition: 0.2s all ease-in;
  }

  span:nth-of-type(1) {
    color: #198754 !important;
  }

  span:nth-of-type(2) {
    text-transform: capitalize !important;
  }

  span:nth-of-type(3) {
    margin-top: 6.4px;
    font-size: 15px;
    color: #757575 !important;
    margin-left: 6px !important;
  }
  
  .entry-description {
    font-size: 13.3px !important;
  }
}


</style>