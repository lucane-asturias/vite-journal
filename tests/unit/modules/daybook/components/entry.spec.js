import { describe, expect, test, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalMockState } from '../../../mock-data/test-journal-state'

describe('Provas no Entry Component', () => {

  const mockRouter = {
    push: vi.fn()
  }

  const wrapper = shallowMount(Entry, {
    props: { entry: journalMockState.entries[0] },
    global: {
      mocks: {
        $router: mockRouter
      }
    }
  })

  test('deve fazer match com o snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('deve redirecionar ao clicar para entry-container', () => {
    const entryContainer = wrapper.find('.entry-container')
    entryContainer.trigger('click')

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry', 
      params: {
        id: journalMockState.entries[0].id
      }
    })
  })

  test('provas nas propriedades computadas', () => {
    expect(wrapper.vm.entryDay).toBe(6)
    expect(wrapper.vm.entryMonth).toBe('novembro')
    expect(wrapper.vm.entryYear).toBe('2022, domingo')
  })

})
