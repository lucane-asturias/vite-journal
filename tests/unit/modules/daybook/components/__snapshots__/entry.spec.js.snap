// Vitest Snapshot v1

exports[`Provas no Entry Component > deve fazer match com o snapshot 1`] = `
"<div class=\\"entry-container mb-3 pointer p-2\\" data-v-4a9e2957=\\"\\">
  <div class=\\"entry-title d-flex\\" data-v-4a9e2957=\\"\\"><span class=\\"text-success fs-5 fw-bold\\" data-v-4a9e2957=\\"\\">6</span><span class=\\"mx-1 fs-5\\" data-v-4a9e2957=\\"\\">novembro</span><span class=\\"mx-2 text-muted mt-1\\" data-v-4a9e2957=\\"\\">2022, domingo</span></div>
  <div class=\\"entry-description mt-1\\" data-v-4a9e2957=\\"\\">Hola mundo desde pruebas xD</div>
</div>"
`;
